### PR TITLE
fix(cli): CLI can't be used in Lambda Function

### LINKED ISSUE #7530: If no home directory exist on host, the CDK will fail
The CDK creates an account cache in: 

```
~/.cdk/cache
```
When running in an environment that does no have a home directory, such as aws-lambda, attempting to create the cache will fail `synth`/`deploy`/`diff`.

### Reproduction Steps
Execute the CDK inside a lambda which defines AWS credentials using AWS environment variables.

### Error Log

```
Toolkit stack: CDKToolkit
Setting "CDK_DEFAULT_REGION" environment variable to us-west-2
Resolving default credentials
Looking up default account ID from STS
Default account ID: xxxxx
EACCES: permission denied, mkdir '/home/xxxx'
Error: EACCES: permission denied, mkdir '/home/xxx'
2020-04-23T03:11:28.801Z        fc4cb3ae-5a84-1734-2972-7f101accf659    INFO    Error: child exited with code 1
    at ChildProcess.<anonymous> (/var/task/node_modules/await-spawn/index.js:28:21)
    at ChildProcess.emit (events.js:311:20)
    at ChildProcess.EventEmitter.emit (domain.js:482:12)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:275:12) {
  code: 1,
  stderr: ''
}
```

### Environment

  - **CLI Version      :** 1.34
  - **Framework Version:** 1.34
  - **OS               :** 
  - **Language         :** typescript

### Other

A way around this is setting the `HOME` environment variable to a directory which exist, and the user has permission on (the user running `cdk`), courtesy of @rhboyd:
```
const main = async () => {
  process.env.HOME = '/tmp';
  const bl = await spawn('node', ['node_modules/aws-cdk/bin/cdk.js', '--app', '"node dist/sample-app.js"', 'synth', '--verbose', '--output', '/tmp', '--no-staging'], { stdio: ['pipe', 'pipe', process.stderr] });
  console.log(bl.toString())
}
```

### PR DESCRIPTION
**Pull request by rix0rrr**
### Commit Message
fix(cli): CLI can't be used in Lambda Function

Two reasons:

* The `CDK_HOME` variable wasn't being consistently honored.
* Failing to write a cache file was a fatal error.

Rectify both of these issues.

Fixes #7530.
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk/lib/api/aws-auth/account-cache.ts [hunk 1]
```diff
@@ -77,18 +77,25 @@ export class AccountAccessKeyCache {
   }
 
   private async loadMap(): Promise<{ [accessKeyId: string]: Account }> {
-    if (!(await fs.pathExists(this.cacheFile))) {
-      return { };
+    try {
+      return await fs.readJson(this.cacheFile);
+    } catch (e) {
+      if (e.code === 'ENOENT' || e.code === 'EACCES') { return {}; }
```
Comment (shivlaks) [inline packages/aws-cdk/lib/api/aws-auth/account-cache.ts:83]: probably worth a comment for posterity

File: packages/aws-cdk/lib/api/aws-auth/account-cache.ts [hunk 2]
```diff
@@ -77,18 +77,25 @@ export class AccountAccessKeyCache {
...
 
   private async saveMap(map: { [accessKeyId: string]: Account }) {
-    if (!(await fs.pathExists(this.cacheFile))) {
-      await fs.mkdirs(path.dirname(this.cacheFile));
-    }
+    try {
+      const dir = path.dirname(this.cacheFile);
+      if (!(await fs.pathExists(dir))) {
+        await fs.mkdirs(dir);
+      }
```
Comment (shivlaks) [inline packages/aws-cdk/lib/api/aws-auth/account-cache.ts:93]: why not something like `fs.ensureFileSync(this.cacheFile);` which will create the directories if it doesn't exist

File: packages/aws-cdk/test/account-cache.test.ts [hunk 3]
```diff
@@ -89,3 +110,28 @@ test(`cache is nuked if it exceeds ${AccountAccessKeyCache.MAX_ENTRIES} entries`
     await nukeCache(cacheDir);
   }
 });
+
+function withMocked<A extends object, K extends keyof A, B>(obj: A, key: K, block: (fn: jest.Mocked<A>[K]) => B): B {
```
Comment (shivlaks) [inline packages/aws-cdk/test/account-cache.test.ts:114]: this is fun :)

File: packages/aws-cdk/lib/api/aws-auth/account-cache.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/aws-cdk/lib/api/aws-auth/account-cache.ts:93]: Sure.

File: packages/aws-cdk/test/account-cache.test.ts (hunk 3 above)
Comment (rix0rrr) [reply packages/aws-cdk/test/account-cache.test.ts:114]: And maybe a little overkill, but may come in useful in other cases as well... remember that this exists (somewhere) !~

**Pull request merged**